src/extension.ts

```diff
@@ -87,17 +97,58 @@ export async function activate(context: ExtensionContext) {
     .then(javaHome => fetchAndLaunchMetals(context, javaHome))
     .catch(err => {
       const message =
-        "Unable to find a Java 8 or Java 11 installation on this computer. " +
-        "To fix this problem, update the 'Java Home' setting to point to a Java 8 or Java 11 home directory";
+        getMessage(err) +
+        "To fix this problem, update the 'Java Home' setting to point to a Java 8 or Java 11 home directory " +
+        "or select version to install automatically (window will reload afterwards)";
+
       outputChannel.appendLine(message);
-      outputChannel.appendLine(err);
-      window.showErrorMessage(message, openSettingsAction).then(choice => {
-        if (choice === openSettingsAction) {
+      if (
+        !(
+          err.message.includes("JavaHome") ||
+          err.message.includes(unsupportedJavaVersion9)
+        )
+      )
+        outputChannel.appendLine(err);
+
+      const jabba_ver = defaultJabbaVer;
+      showJabbaMessage(message, jabba_ver);
+    });
+  commands.executeCommand("setContext", "metals:enabled", true);
+}
+
+function showJabbaMessage(
+  message: string,
+  jabba_ver: string = defaultJabbaVer
+) {
+  return window
+    .showErrorMessage(message, openSettingsAction, jdk8Action, jdk11Action)
+    .then(choice => {
+      const jdkinstaller = new JdkInstaller(outputChannel);
+      switch (choice) {
+        case openSettingsAction: {
           commands.executeCommand("workbench.action.openSettings");
+          break;
         }
-      });
+        case jdk8Action: {
+          window.showInformationMessage(`Installing ${jdk8}, please wait...`);
```

Comment:
Can we use `window.withProgress` here? Tested it out and it looks really cool, the only issue I have is the window just shows up and it's hard to say what is going on, it would be nice to have:
1. `Installing ${jdk8}, please wait...` showing progress (or just 100% when it's done)
2. Then the window disappears and reload suggestion appears.

Reply:
Ok, that's fixed and it shows progress nicely